lib/rubygems/commands/query_command.rb

```diff
@@ -78,46 +78,58 @@ def description # :nodoc:
   end
 
   def execute
+    gem_names = Array(options[:name])
+
+    if !args.empty?
+      gem_names = options[:exact] ? args.map{|arg| /\A#{Regexp.escape(arg)}\Z/ } : args.map{|arg| /#{arg}/i }
+    end
+
+    terminate_interaction(check_installed_gems(gem_names)) if check_installed_gems?
+
+    gem_names.each { |n| show_gems(n) }
+  end
+
+  private
+
+  def check_installed_gems(gem_names)
     exit_code = 0
-    if options[:args].to_a.empty? and options[:name].source.empty?
-      name = options[:name]
-      no_name = true
-    elsif !options[:name].source.empty?
-      name = Array(options[:name])
+
+    if args.empty? && !gem_name?
+      alert_error "You must specify a gem name"
+      exit_code = 4
+    elsif gem_names.count > 1
+      alert_error "You must specify only ONE gem!"
+      exit_code = 4
     else
-      args = options[:args].to_a
-      name = options[:exact] ? args.map{|arg| /\A#{Regexp.escape(arg)}\Z/ } : args.map{|arg| /#{arg}/i }
+      installed = installed?(gem_names.first, options[:version])
+      installed = !installed unless options[:installed]
+
+      say(installed)
+      exit_code = 1 if !installed
     end
 
-    prerelease = options[:prerelease]
+    exit_code
+  end
 
-    unless options[:installed].nil?
-      if no_name
-        alert_error "You must specify a gem name"
-        exit_code |= 4
-      elsif name.count > 1
-        alert_error "You must specify only ONE gem!"
-        exit_code |= 4
-      else
-        installed = installed? name.first, options[:version]
-        installed = !installed unless options[:installed]
+  def check_installed_gems?
+    !options[:installed].nil?
+  end
 
-        if installed
-          say "true"
-        else
-          say "false"
-          exit_code |= 1
-        end
-      end
+  def gem_name?
+    !options[:name].source.empty?
+  end
 
-      terminate_interaction exit_code
-    end
+  def prerelease?
+    options[:prerelease]
+  end
 
-    names = Array(name)
-    names.each { |n| show_gems n, prerelease }
+  def show_prereleases?
+    options[:prerelease].nil? || prerelease?
```

Comment:
You can use the `prerelease` method in the first part of the condition as well :)

Reply:
isn't `prerelease?.nil?` to weird?

Reply:
Hahaha, you're right! Maybe we can call the method `prerelease` instead, since it's not strictly boolean anyways.

Reply:
Fair 👍